src/pybind/mgr/ansible/module.py

```diff
@@ -328,44 +330,49 @@ def process_inventory_json(inventory_events, ar_client, playbook_uuid):
         Example:
         inventory_events =
         {'37-100564f1-9fed-48c2-bd62-4ae8636dfcdb': {'host': '192.168.121.254',
-                                                    'task': 'RESULTS',
+                                                    'task': 'list storage inventory',
                                                     'event': 'runner_on_ok'},
         '36-2016b900-e38f-7dcd-a2e7-00000000000e': {'host': '192.168.121.252'
-                                                    'task': 'RESULTS',
+                                                    'task': 'list storage inventory',
                                                     'event': 'runner_on_ok'}}
     @param ar_client: Ansible Runner Service client
     @param playbook_uuid: Palybooud identifier
 
     @return              : list of InventoryNode
+
+
     """
 
     #Obtain the needed data for each result event
     inventory_nodes = []
 
     # Loop over the result events and request the event data
-    for event_key, data in inventory_events.items():
+    for event_key, dummy_data in inventory_events.items():
+
         event_response = ar_client.http_get(EVENT_DATA_URL % (playbook_uuid,
                                                               event_key))
 
         # Process the data for each event
         if event_response:
             event_data = json.loads(event_response.text)["data"]["event_data"]
 
-            free_disks = event_data["res"]["disks_catalog"]
-            for item, data in free_disks.items():
-                if item not in [host.name for host in inventory_nodes]:
-
-                    devs = []
-                    for dev_key, dev_data in data.items():
-                        if dev_key not in [device.id for device in devs]:
-                            dev = orchestrator.InventoryDevice()
-                            dev.id = dev_key
-                            dev.type = 'hdd' if dev_data["rotational"] else "sdd/nvme"
-                            dev.size = dev_data["sectorsize"] * dev_data["sectors"]
-                            devs.append(dev)
-
-                    inventory_nodes.append(
-                            orchestrator.InventoryNode(item, devs))
+            host = event_data["host"]
+            devices =  json.loads(event_data["res"]["stdout"])
+
+            devs = []
+            for storage_device in devices:
+                dev = orchestrator.InventoryDevice()
+                dev.id = storage_device["path"]
+                dev.type = 'hdd' if storage_device["sys_api"]["rotational"]== "1" else "sdd/nvme"
```

Comment:
```suggestion
                dev.type = 'hdd' if storage_device["sys_api"]["rotational"] == "1" else 'sdd/nvme'
```

Reply:
Fixed. Thanks! (More coherent in this way)